### PR TITLE
Add replicas of default build configs with OpenJDK JSR292 enabled

### PR DESCRIPTION
**Pull request by babsingh**
The default build configs are the ones with `no suffix` and `_xl` suffix.

The default build configs are replicated to create new build configs
with OpenJDK JSR292 enabled.

OpenJDK JSR292 is enabled using the following configure option:
`--enable-openjdk-methodhandles`.

The new build configs will be used to test the support for OpenJDK
JSR292 until it is confirmed to be stable. After confirmation of
stability, the support for OpenJDK JSR292 will be enabled in the default
configs, and the new build configs will be deleted.

#11116 tracks the deletion of new build configs.

Added a missing shortname for zOS: `zoslargeheap`.

Related: https://github.com/eclipse/openj9/issues/7352

Co-authored-by: Jack Lu <Jack.S.Lu@ibm.com>
Signed-off-by: Babneet Singh <sbabneet@ca.ibm.com>

### CODE REVIEW

**Comment by babsingh** [comment]
@AdamBrousseau I just added one new configuration with OpenJDK JSR292 enabled. Can you provide feedback on the changes? After receiving your feedback, I will extend it to all platforms.

`'x86-64_linux_ojdk292' : []`: Does this mean that the configuration is not supported for any release?

What will be the command line for running this configuration as part of PR builds?

**Comment by AdamBrousseau** [comment]
Jenkins compile xlinuxojdk292 jdknext

**Comment by babsingh** [comment]
Few PRs are still pending to be merged. So, the PR build will fail.

**Comment by babsingh** [comment]
fyi @DanHeidinga @tajila @fengxue-IS @pshipton @andrewcraik

**Comment by AdamBrousseau** [comment]
>Does this mean that the configuration is not supported for any release?

Correct. In that case I don't see the point of adding the line.

**Review by AdamBrousseau — APPROVED** [review]

**Comment by babsingh** [comment]
@AdamBrousseau Will merging these changes enable nightly builds? If nightly builds won't be enabled by these changes, then these changes can be merged before OpenJDK MethodHandles are fully functional.

**Comment by AdamBrousseau** [comment]
>Will merging these changes enable nightly builds? 

No, all builds specify what platforms to build so it won't start showing up somewhere unintended. It will allow devs to do personal builds or PR builds with the spec.

**Comment by babsingh** [comment]
@AdamBrousseau Can you please review this PR?

The new configs will be eventually deleted once stability is confirmed for the support of OpenJDK JSR292. I will create an issue to delete the new configs after this PR is approved and merged.

**Comment by babsingh** [comment]
@AdamBrousseau What's the process for setting up nightly builds for any of the new build specs?

**Comment by AdamBrousseau** [comment]
If you just want specs added to the existing nightlies, running the same amount of testing, we can add them to the regular builds. If you want a subset of testing we could add a new build specific to the 292 specs and have more control over what testing is run.
Pete or myself can make the change (jenkins UI) once we have decided.

File: buildenv/jenkins/jobs/pipelines/Pipeline-Build-Test-All.groovy
Comment (AdamBrousseau) [inline buildenv/jenkins/jobs/pipelines/Pipeline-Build-Test-All.groovy:210]: Seems to be a non-292 change here. I guess you're just adding a missed shortname for zos.

**Review by AdamBrousseau — APPROVED** [review]
lftm

File: buildenv/jenkins/jobs/pipelines/Pipeline-Build-Test-All.groovy
Comment (babsingh) [reply buildenv/jenkins/jobs/pipelines/Pipeline-Build-Test-All.groovy:210]: Yes, that is correct. Would you like it in a separate commit?

Comment (AdamBrousseau) [reply buildenv/jenkins/jobs/pipelines/Pipeline-Build-Test-All.groovy:210]: No that's fine. It just caught my eye so I was trying to figure it out.

**Comment by babsingh** [comment]
@DanHeidinga @pshipton This needs to be tagged with the `project:MH` label. #11116 tracks the deletion of new build configs. This PR is ready for a **committer review**.

**Comment by pshipton** [comment]
jenkins compile zlinux jdk8

**Comment by pshipton** [comment]
```
WorkflowScript: 235: expecting ']', found 'zoslargeheapojdk292' @ line 235, column 13.
               'zoslargeheapojdk292' : ['s390x_zos_xl_ojdk292']]
```

**Comment by pshipton** [comment]
missing `,` on the previous line.

**Comment by babsingh** [comment]
> missing `,` on the previous line.

The missing `,` has been added on line `234` of `Pipeline-Build-Test-All.groovy`.

**Comment by pshipton** [comment]
jenkins compile zlinux jdk8

**Review by pshipton — APPROVED** [review]

**Pull request merged**
